num_enum_derive/src/lib.rs

```diff
@@ -489,7 +554,11 @@ pub fn derive_into_primitive(input: TokenStream) -> TokenStream {
 #[proc_macro_derive(FromPrimitive, attributes(num_enum, default, catch_all))]
 pub fn derive_from_primitive(input: TokenStream) -> TokenStream {
     let enum_info: EnumInfo = parse_macro_input!(input);
-    let krate = Ident::new(&get_crate_name(), Span::call_site());
+    let krate = enum_info.crate_path.clone().unwrap_or_else(|| {
+        syn::LitStr::new(&format!("::{}", get_crate_name()), Span::call_site())
+            .parse_with(syn::Path::parse_mod_style)
+            .unwrap()
```

Comment:
```suggestion
        let krate = get_crate_name();
        parse_quote!( :: #krate )
```

Reply:
this `parse_quote!` would generate `:: "num_enum"` because krate is a `String`

Reply:
maybe `let krate = format_ident!("{}", get_crate_name());`?

Reply:
Ah true, yeah what you've just suggested would be the right thing.

Although at that point, I guess we could go back to something closer to your original suggestion:

```rs
syn::parse_str(&format!("::{}", get_crate_name())).unwrap()
```

  - (it just sidesteps the intermediary `LitStr`)